linode/image/framework_models.go

```diff
@@ -3,11 +3,94 @@ package image
 import (
 	"time"
 
+	"github.com/hashicorp/terraform-plugin-framework-timeouts/resource/timeouts"
+	"github.com/hashicorp/terraform-plugin-framework-timetypes/timetypes"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/types"
 	"github.com/linode/linodego"
 	"github.com/linode/terraform-provider-linode/v2/linode/helper"
 )
 
+type ResourceModel struct {
+	ID           types.String      `tfsdk:"id"`
+	Label        types.String      `tfsdk:"label"`
+	DiskID       types.Int64       `tfsdk:"disk_id"`
+	LinodeID     types.Int64       `tfsdk:"linode_id"`
+	FilePath     types.String      `tfsdk:"file_path"`
+	Region       types.String      `tfsdk:"region"`
+	FileHash     types.String      `tfsdk:"file_hash"`
+	Description  types.String      `tfsdk:"description"`
+	CloudInit    types.Bool        `tfsdk:"cloud_init"`
+	Capabilities types.List        `tfsdk:"capabilities"`
+	Created      timetypes.RFC3339 `tfsdk:"created"`
+	CreatedBy    types.String      `tfsdk:"created_by"`
+	Deprecated   types.Bool        `tfsdk:"deprecated"`
+	IsPublic     types.Bool        `tfsdk:"is_public"`
+	Size         types.Int64       `tfsdk:"size"`
+	Status       types.String      `tfsdk:"status"`
+	Type         types.String      `tfsdk:"type"`
+	Expiry       timetypes.RFC3339 `tfsdk:"expiry"`
+	Vendor       types.String      `tfsdk:"vendor"`
+	Timeouts     timeouts.Value    `tfsdk:"timeouts"`
+}
+
+func (data *ResourceModel) FlattenImage(image *linodego.Image, preserveKnown bool, diags *diag.Diagnostics) {
+	data.ID = helper.KeepOrUpdateString(data.ID, image.ID, preserveKnown)
+	data.Label = helper.KeepOrUpdateString(data.Label, image.Label, preserveKnown)
+	data.Description = helper.KeepOrUpdateString(
+		data.Description, image.Description, preserveKnown,
+	)
+
+	newCapabilities, newDiags := types.ListValue(
+		types.StringType, helper.StringSliceToFrameworkValueSlice(image.Capabilities),
+	)
+	diags.Append(newDiags...)
+	if diags.HasError() {
+		return
+	}
+
+	data.Capabilities = helper.KeepOrUpdateValue(data.Capabilities, newCapabilities, preserveKnown)
+	// TODO: should we do slice in model? Will it work? need more exploration here
```

Comment:
I think keep it as list would be fine. With a slice of `types.String`, it will be harder to determine whether it's a Unknown value that can be safely overridden.